ldapauthenticator/ldapauthenticator.py

```diff
@@ -556,11 +556,19 @@ async def authenticate(self, handler, data):
         return {"name": username, "auth_state": auth_state}
 
     async def check_allowed(self, username, auth_model):
-        allowed = super().check_allowed(username, auth_model)
-        if isawaitable(allowed):
-            allowed = await allowed
-        if allowed is True:
-            return True
+        if not hasattr(self, "allow_all"):
+            # super for JupyterHub < 5
+            # default behavior: no allow config => allow all
+            if not self.allowed_users and not self.allowed_groups:
+                return True
```

Comment:
I think this maybe should include admin_users as well, as configuring that is a way to allow users access.

Reply:
I don't think so. `admin_users` is not checked for access, but it will populate `allowed_users` if it is not empty during JupyterHub startup. So `admin_users` is already contained within `allowed_users`.

Reply:
[this is the behavior we want to preserve](https://github.com/jupyterhub/jupyterhub/blob/4.1.6/jupyterhub/auth.py#L447-L450), which does not reference `admin_users`.